### PR TITLE
Development branch, release 3.3.0

### PR DESCRIPTION
**Pull request by AlexanderFabisch**
## Features

* Add option to visualize convex hull of collision objects.
* Add `TemporalTransformManager` to handle dynamic transformations.

## Documentation

* New logo.
* Add section on temporal alignment of transformations.

## Refactoring

* Remove dependency beautifulsoup4 and use lxml for URDF parsing.
* Turn `pytransform3d.transform_manager` into a subpackage.
* Extract base class `TransformGraphBase` from `TransformManager` that contains common code of `TransformManager` and `TemporalTransformManager`.

### CODE REVIEW

**Comment by AlexanderFabisch** [comment]
@kopytjuk Thanks for the contribution of the TemporalTransformManager! From my perspective it is done. Please have a look at the final version and confirm that it works for you. I would like to release version 3.3.0 soon.

File: examples/plots/plot_interpolation_for_transform_manager.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+
+from pytransform3d import rotations as pr
+from pytransform3d import transformations as pt
+from pytransform3d.transform_manager import (TemporalTransformManager,
+                                             NumpyTimeseriesTransform)
+
+
+def create_sinusoidal_movement(
```
Comment (kopytjuk) [inline examples/plots/plot_interpolation_for_transform_manager.py:18]: Consider moving it to a utility module (which is excluded from the python wheel) but still can be usable in example plots or documentation - here we have a real duplication of things...

File: pytransform3d/test/test_transform_manager.py [hunk 2]
```diff
@@ -4,11 +4,14 @@
 import tempfile
 import numpy as np
 from pytransform3d.rotations import (
```
Comment (kopytjuk) [inline pytransform3d/test/test_transform_manager.py:6]: Would it make sense to import the namespace as `tf3d_rot` and call the functions in the unittest?

File: pytransform3d/test/test_transform_manager.py [hunk 3]
```diff
@@ -333,3 +338,129 @@ def test_remove_and_add_connection():
...
+    tm = TemporalTransformManager()
+
+    tm.add_transform("A", "B", StaticTransform(A2B))
+    tm.add_transform("A", "C", StaticTransform(A2C))
+
+    tm.remove_transform("A", "C")
+    assert ("A", "C") not in tm.transforms
+
+
+def create_sinusoidal_movement(
```
Comment (kopytjuk) [inline pytransform3d/test/test_transform_manager.py:382]: The duplicate from the example... See comment above

File: pytransform3d/test/test_transform_manager.py [hunk 4]
```diff
@@ -333,3 +338,129 @@ def test_remove_and_add_connection():
...
+        origin_of_A_in_B_xyz[0], origin_of_A_in_B_xyz[1]
+
+    assert origin_of_A_in_B_x == pytest.approx(-1.11, abs=1e-2)
+    assert origin_of_A_in_B_y == pytest.approx(-1.28, abs=1e-2)
+
+
+def test_numpy_timeseries_transform_wrong_input_shapes():
+    n_steps = 10
+    with pytest.raises(
+            ValueError, match="Number of timesteps does not equal to number "
```
Comment (kopytjuk) [inline pytransform3d/test/test_transform_manager.py:451]: I would just match for a substring, so the essence of the error is tested, such as "number of timesteps" so we can still correct spelling or verbosity of the message... But this is probably subjective

File: pytransform3d/transform_manager/_temporal_transform_manager.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+from ..batch_rotations import norm_vectors
+
+from ._transform_graph_base import TransformGraphBase
+from ..transformations import check_transform, transform_from_pq, \
+    dual_quaternion_from_pq, pq_from_dual_quaternion, dual_quaternion_sclerp
+
+
+class TimeVaryingTransform(abc.ABC):
+    """Time-varying rigid transformation."""
```
Comment (kopytjuk) [inline pytransform3d/transform_manager/_temporal_transform_manager.py:13]: Maybe we can add "This class shall be inherited from to use the `TemporalTransformManager`, refer to NumpyTransform implementation" or something similar.

This text appears in the docs, so people browsing around will see what this class is meant for.

**Review by kopytjuk — APPROVED** [review]
LGTM, small comments left... Nothing major

**Comment by AlexanderFabisch** [comment]
@kopytjuk It's done. Thanks a lot for your contribution!

**Pull request merged**